### DISCUSSION TITLE
heavy isotope label

### DISCUSSION THREAD

**Discussion by DanielVogele** [vdemichev/DiaNN#542] [Q&A]
Hi Vadim,

is it possible to search for heavy isotope labels? 
In my case, we have heavy isotope labelled R / K on a certain selection of peptides spiked into a HEK background. We would like to search for these heavy isotope labelled peptides as well as the unlabelled background. This might also need to be considered for the library prediction.

I've read that this might be possible via a command line GUI in DIA-NN. For publication see here:
https://www.nature.com/articles/s41587-022-01389-w#Sec12

"The spectral library used for plexDIA benchmarking was created in a similar process, with the exception of a few additional commands entered into the DIA-NN command line GUI: (1) {–fixed-mod mTRAQ 140.0949630177, nK} and (2) {–original-mods}."

Best,
Daniel

**Comment by vdemichev** [comment]
Hi Daniel,

Yes, it is, using the --channels command, please email me for details on the settings.

Best,
Vadim

**Comment by vincentroyrichard** [comment]
Hi Vadim, 

I would also like to use DIA-NN to search for both K+8, R+10 labelled tryptic peptides. Is there a tutorial somewhere on how to do this via the command line?

Thanks very much,
Vincent

**Comment by vdemichev** [comment]
Please find below the detailed description of how to analyse SILAC data with DIA-NN (v1.8.1):

How to use:
Specify a spectral library  - can be an empirical one or an in silico predicted one. Then add the following commands to the 'Additional options' textbox:
--fixed-mod SILAC,0.0,KR,label
--lib-fixed-mod SILAC
--channels SILAC,L,KR,0:0; SILAC,H,KR,8.014199:10.008269
--peak-translation
--original-mods

--fixed-mod here declares SILAC as a label
--lib-fixed-mod in silico labels the library with it. As masses are 0, it just adds SILAC to precursor names
--channels declares channels. Channel 'L' is declared with masses 0 on K and 0 on R. Channel 'H' with masses 8 on K and 10 on R.
--relaxed-prot-inf - as usual, recommended for benchmarking if the library does not have protein groups assembled properly already.

Report:
Precursor.Translated - special quantity calculated when having multiple channels - most likely the best one. If you need protein quantities, need to calculate those based on Precursor.Translated with MaxLFQ, either using iq R package (recommended) or diann R package.
Channel.Q.Value - probability that a given precursor is indeed present in the channel
Translated.Quality - reflects how much evidence there was for the particular Precursor.Translated (the quantity I suggest to use), that is how many fragments could be use to calculate ratios between channels.

--matrix-ch-qvalue  and --matrix-tr-qvalue set the channel and translated q-value filtering for the output matrices. These commands can be useful if you use the matrices as the output.

**Reply by vincentroyrichard** [reply]
Thanks very much Vadim - I will try it out and report back if there are any issues.

**Reply by vincentroyrichard** [reply]
Hi Vadim,

Just to follow up with this - I was eventually able to get it to work without giving an error, but I think I must have done something wrong because I am getting quantitation for a label which was not present in the sample. 

In my experiment we generated a spectral library from fractionated K+6 labelled peptides (DDA-PASEF). Library generation was performed with Fragpipe v21 . We then spiked labelled peptides into unlabelled background and analyzed by DIA-PASEF. Only K+0/R+0, and K+6 should be present, but I am seeing labelled R+10 in the Precursor.Translated report. Is this something you have observed before? I've attached the log file below.

Thanks very much for developing these tools and for helping us to use them!
Vincent

[01_15_2024 14_24_01.txt](https://github.com/vdemichev/DiaNN/files/13942907/01_15_2024.14_24_01.txt)

**Reply by vdemichev** [reply]
"Only K+0/R+0, and K+6 should be present" - please adjust the commands accordingly, i.e. R should not be declared as part of --var-mod for SILAC

Best,
Vadim

**Reply by khuehua** [reply]
Hi Vadim,

I also did the search for both heavy and light peptides (same as Vincent with K+8 and R+10). 

I added the commands you suggested above to the "Additional options" box (v.2.0.2) and searched against the predicted library generated from DIA-NN. However, it turned out that there is no protein annotation at all and the parquet file seems empty. Do you have any suggestions to overcome that? Thanks in advance!

P/s: I attached the log file in case you would like to look at it!
[blood_test.log.txt](https://github.com/user-attachments/files/19112249/blood_test.log.txt)

**Reply by vdemichev** [reply]
> WARNING: protein inference is enabled but no FASTA provided - is this intended?

Please just provide a FASTA

**Reply by khuehua** [reply]
Hi Vadim,

After providing a FASTA, I still got Number of genes identified at 1% FDR: 35 (precursor-level), 0 (protein-level) (inference performed using proteotypic peptides only). Could you help me to address it? I also attached the log file here. Thanks in advance!

[test.log.txt](https://github.com/user-attachments/files/19279576/test.log.txt)

**Reply by vdemichev** [reply]
Please see the main .parquet report, all the gene info should be there.
